bundles/org.openhab.binding.sleepiq/src/main/java/org/openhab/binding/sleepiq/internal/handler/SleepIQCloudHandler.java

```diff
@@ -262,6 +274,116 @@ public void setPauseMode(@Nullable String bedId, boolean mode) {
         }
     }
 
+    /**
+     * Get the foundation features of the specified bed
+     *
+     * @param bedId the bed identifier
+     */
+    public @Nullable FoundationFeaturesResponse getFoundationFeatures(@Nullable String bedId) {
+        if (bedId == null) {
```

Comment:
It doesn't make sense to call this method with `bedId == null`. Consider making it non-nullable and avoid calling it in this case.

Reply:
Fixed